### ISSUE TITLE
Feature Request: GPU support

### ISSUE THREAD

**Issue by DjangoPeng** [kubernetes/kubernetes#40998]
I'm an active open-source contributor of Kubernetes and TensorFlow. I've built kubernetes v1.5 for managing distributed tensorflow tasks. While what make me upset is that kubernetes doesn't support GPU allocation. 
I can't enjoy the convenient management of kubernetes when it comes to TensorFlow and Deep Learning, because Deep Learning tasks always using GPU instead of CPU. I know [Nvidia-docker](https://github.com/NVIDIA/nvidia-docker) is a solution for that.
 Do you have a roadmap for supporting GPU in kubernetes v1.6?

**Comment by k82cn** [comment]
AFAIK, our research team has a PR for GPU; let me find it :).

**Comment by k82cn** [comment]
@liyubobj , is it possible to share the PR about GPU?

**Comment by cmluciano** [comment]
@k82cn @DjangoPeng Initial PR is here https://github.com/kubernetes/kubernetes/pull/28216. I am also preparing a follow-up PR to add NVML support soon. I think the PR linked is still targeted for 1.6.

**Comment by liyubobj** [comment]
@DjangoPeng, @Hui-Zhi is working on the prelimiary GPU support, please follow #28216 for latest updates. Based on this PR, we also developed more advanced features waiting to merge to upstream, including GPU driver volume injection (same as nvidia-docker), heterogeneous GPU support, GPU topology-aware scheduling, etc. The timeline depends on #28216 , I'm sorry that I can not give you detailed roadmap.

**Comment by liyubobj** [comment]
agree with @cmluciano :)

**Comment by Hui-Zhi** [comment]
Yes, #28216 is targeted for 1.6

**Comment by DjangoPeng** [comment]
@Hui-Zhi Thanks for your reply and work around GPU support. I'd like to try it in RHEL 7.0. Can we get the feature in kubernetes 1.6.0 release version?

**Comment by Hui-Zhi** [comment]
According to @vishh , yes.

**Comment by davidopp** [comment]
FWIW there are already several other issues for GPU support -- can we consolidate down to one (or at least fewer than five)?
ref/ #17035 #19049 #23587 #25557

**Comment by Hui-Zhi** [comment]
@davidopp I have closed #25557 and #19049 , but I don't have the privilege to close the others. Please take a look at #28216 if you have time.

**Comment by ahundt** [comment]
https://github.com/kubernetes/kubernetes/pull/42116 I followed the chain and this is the current PR for GPU support.

**Comment by cmluciano** [comment]
@DjangoPeng #42116 is now merged and should be released with 1.6

**Comment by cmluciano** [comment]
This should be closed now that alpha support is merged in 1.6

/assign /close

**Comment by DjangoPeng** [comment]
@cmluciano Thanks, I'm trying the GPU feature in 1.7.0-alpha4

**Comment by cmluciano** [comment]
Thanks @DjangoPeng. We have https://github.com/kubernetes/kubernetes/issues/44107 open to further our support.
